File: kotlin/src/main/kotlin/common/infrastructure/stream/FileMessageBroker.kt
package common.infrastructure.stream

import common.infrastructure.stream.producer.SerializableEvent
import java.io.File
import java.io.IOException

class FileMessageBroker(private val filePath: String): MessageBroker {
    override fun publish(event: SerializableEvent, topic: String) {
        val file = File("$filePath$topic.events")
        try {
            file.appendText(event.toJson() + System.lineSeparator())
        } catch (e: IOException) {
            throw IOException("Failed to write to stream", e)
        }
    }
}

File: typescript/src/common/infrastructure/stream/producer/FileMessageBroker.ts
import {SerializableEvent} from "../SerializableEvent";
import {MessageBroker} from "../MessageBroker";

export class FileMessageBroker implements MessageBroker {
    private filePath: string;

    constructor(filePath: string) {
        this.filePath = filePath;
    }

    /**
     * @throws Error
     */
    public publish(event: SerializableEvent, topic: string): void {
        const fs = require('fs');
        const path = `${this.filePath}${topic}.events`;
        const data = event.toJson() + '\n';

        try {
            fs.appendFileSync(path, data);
        } catch (err) {
            throw new Error('Failed to write to stream');
        }
    }
}
